packages/frontend-v2/components/Sidebar/Sidebar.tsx

```diff
@@ -115,79 +113,34 @@ const Sidebar: React.FC<SidebarProps> = memo(({ selectedPlan }) => {
           default:
             throw new Error("Invalid worker message!");
         }
-      }
+      };
     }
     return () => {
       workerRef.current?.terminate();
       workerRef.current = undefined;
-    }
-  // LINT NOTE: We don't actually want a dependency to the local function 
-  // revalidateMajor because it will change every time, so we're choosing 
-  // to omit it here:
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [])
-
+    };
+    // LINT NOTE: We don't actually want a dependency to the local function
+    // revalidateMajor because it will change every time, so we're choosing
+    // to omit it here:
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
   // Whenever our plan/major changes, we revalidate if the worker
   // is initialized.
-  // LINT NOTE: We don't actually want a dependency to the local function 
-  // revalidateMajor because it will change every time, so we're choosing 
+  // LINT NOTE: We don't actually want a dependency to the local function
+  // revalidateMajor because it will change every time, so we're choosing
   // to omit it here:
   // eslint-disable-next-line react-hooks/exhaustive-deps
-  useEffect(() => revalidateMajor(), [selectedPlan, major])
-
-  const [courseData, setCourseData] = useState({});
-  const [loading, setLoading] = useState(true);
+  useEffect(() => revalidateMajor(), [selectedPlan, major]);
 
-  // Get course names/data from SearchNEU's API
-  useEffect(() => {
-    if (!major) {
-      return;
-    }
+  const majorCourses = getAllCoursesInMajor(major, concentration);
 
-    const concentrationRequirements: IRequiredCourse[] = [];
-    getRequiredCourses(
-      concentration?.requirements ?? [],
-      concentrationRequirements
-    );
+  const { courses, isLoading: isCoursesLoading } =
```

Comment:
error state here is very important: essentially handling the state where search doesn't return any courses.